server/app/modules/SecurityModule.java

```diff
@@ -81,44 +78,16 @@ protected void configure() {
     logoutController.setDefaultUrl(baseUrl + routes.HomeController.index().url());
     logoutController.setLocalLogout(true);
     logoutController.setDestroySession(true);
-    /*
-     * This mitigates a vulnerability in the logout process described here:
-     * https://groups.google.com/g/pac4j-security/c/poWGfZKo-ww/m/S-h4ggaSAgAJ
-     *
-     * <p>Can be removed after upgrading to pac4j v5.6.1
-     */
-    logoutController.setLogoutUrlPattern("^(\\/|\\/[^\\/].*)$");
 
     Boolean shouldPerformAuthProviderLogout = configuration.getBoolean("auth.oidc_provider_logout");
     logoutController.setCentralLogout(shouldPerformAuthProviderLogout);
     bind(LogoutController.class).toInstance(logoutController);
 
-    // This is a weird one.  :)  The cookie session store refuses to serialize any
-    // classes it doesn't explicitly trust.  A bug in pac4j interacts badly with
-    // sbt's autoreload, so we have a little workaround here.  configure() gets called on every
-    // startup,
-    // but the JAVA_SERIALIZER object is only initialized on initial startup.
-    // So, on a second startup, we'll add the CiviFormProfileData a second time.  The
-    // trusted classes set should dedupe CiviFormProfileData against the old CiviFormProfileData,
-    // but it's technically a different class with the same name at that point,
-    // which triggers the bug.  So, we just clear the classes, which will be empty
-    // on first startup and will contain the profile on subsequent startups,
-    // so that it's always safe to add the profile.
-    // We will need to do this for every class we want to store in the cookie.
-    PlayCookieSessionStore.JAVA_SERIALIZER.clearTrustedClasses();
-    PlayCookieSessionStore.JAVA_SERIALIZER.addTrustedClass(CiviFormProfileData.class);
-
-    // We need to use the secret key to generate the encrypter / decrypter for the
-    // session store, so that cookies from version n of the application can be
-    // read by version n + 1.  This is especially important for dev, otherwise
-    // we're going to spend a lot of time deleting cookies.
-    Random r = new Random();
-    r.setSeed(this.configuration.getString("play.http.secret.key").hashCode());
-    byte[] aesKey = new byte[32];
-    r.nextBytes(aesKey);
-    PlayCookieSessionStore sessionStore =
-        new PlayCookieSessionStore(new ShiroAesDataEncrypter(aesKey));
-    bind(SessionStore.class).toInstance(sessionStore);
```

Comment:
All this still exists, but is in a separate class now that it needs to be injected elsewhere.